controllers/cartController.js

```diff
@@ -0,0 +1,63 @@
+const cartService = require('../services/cartService');
+const { catchError } = require('../middlewares/error');
+
+const createCartItem = catchError(async (req, res) => {
+  try {
+    const { productId, quantity } = req.body;
+
+    const userId = req.user.id;
+
+    const cartItem = await cartService.createCartItem({
+      userId,
+      productId,
+      quantity,
+    });
+    res.status(201).json({ message: cartItem });
+  } catch (err) {
+    res.status(400).json({ message: err.message });
+  }
+});
```

Comment:
```suggestion
const createCartItem = catchError(async (req, res) => {
  try {
    const { productId, quantity } = req.body;

    const userId = req.user.id;

    const cartItem = await cartService.createCartItem({
      userId,
      productId,
      quantity,
    });
    res.status(201).json({ message: cartItem });
  } catch (err) {
    res.status(400).json({ message: err.message });
  }
});
```
catchError 라는 에러핸들링에 관한 함수의 코드를 보면, 해당 함수의 인자로 들어가는 함수, 즉 controller 함수에는 try catch가 굳이 필요하지 않습니다.